### PR TITLE
NRFX-4127/8/9 Charger die temperature shell commands

### PR DESCRIPTION
**Pull request by jaz1-nordic**

### CODE REVIEW

File: drivers/npmx/npmx_shell.c
Comment (wilu-nordic) [inline drivers/npmx/npmx_shell.c:1]: `cmd_die_temp_resume_get` and `cmd_die_temp_stop_get` functions are almost the same, function bodies should be merged into one part.

The same applies to `..._set` pair.

**Review by annakielar — APPROVED** [review]

File: drivers/npmx/npmx_shell.c [hunk 1]
```diff
@@ -520,6 +520,182 @@ static int cmd_charger_trickle_set(const struct shell *shell, size_t argc, char
...
+	if ((modules_mask & NPMX_CHARGER_MODULE_CHARGER_MASK) != 0) {
+		shell_error(shell, "Error: charger must be disabled to set threshold value.");
+		return 0;
+	}
+
+	err_code = npmx_charger_die_temp_stop_set(charger_instance, temperature);
+	if (check_error_code(shell, err_code)) {
+		shell_print(shell, "Success: %d *C.", temperature);
+	} else {
+		shell_error(shell, "Error: unable to read die temperature stop threshold.");
```
Comment (magp-nordic) [inline drivers/npmx/npmx_shell.c:661]:
```suggestion
		shell_error(shell, "Error: unable to set die temperature stop threshold.");
```

File: drivers/npmx/npmx_shell.c [hunk 2]
```diff
@@ -1667,20 +1843,56 @@ SHELL_STATIC_SUBCMD_SET_CREATE(sub_charger_trickle,
...
+			       SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 3 command) array for command "charger die_temp". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_die_temp,
+			       SHELL_CMD(resume, &sub_die_temp_resume,
+					 "Die temperature threshold where charging resumes", NULL),
+			       SHELL_CMD(stop, &sub_die_temp_stop,
+					 "Die temperature threshold where charging stops", NULL),
+					SHELL_CMD(status, &sub_die_temp_status,
+					 "Die temperature threshold where charging stops", NULL),
```
Comment (mkleszcz-no) [inline drivers/npmx/npmx_shell.c:1880]:
```suggestion
			       SHELL_CMD(stop, &sub_die_temp_stop,
					 "Die temperature threshold where charging stops", NULL),
				SHELL_CMD(status, &sub_die_temp_status,
					 "Die temperature threshold where charging stops", NULL),
```
Looks like lack of indentation.

**Review by annakielar — APPROVED** [review]

**Review by wilu-nordic — APPROVED** [review]

File: drivers/npmx/npmx_shell.c [hunk 3]
```diff
@@ -1667,20 +1794,55 @@ SHELL_STATIC_SUBCMD_SET_CREATE(sub_charger_trickle,
...
+	sub_charger,
+	SHELL_CMD(termination_voltage, &sub_charger_termination_voltage,
+		  "Charger termination voltage", NULL),
+	SHELL_CMD(termination_current, &sub_charger_termination_current,
+		  "Charger termination current", NULL),
+	SHELL_CMD(charger_current, &sub_charger_charging_current, "Charger current", NULL),
+	SHELL_CMD(status, &sub_charger_status, "Charger status", NULL),
+	SHELL_CMD(module, &sub_charger_module, "Charger module", NULL),
+	SHELL_CMD(trickle, &sub_charger_trickle, "Charger trickle voltage", NULL),
+	SHELL_CMD(die_temp, &sub_die_temp, "Charger die temperature", NULL), SHELL_SUBCMD_SET_END);
```
Comment (masz-nordic) [inline drivers/npmx/npmx_shell.c:1844]:
```suggestion
	SHELL_CMD(die_temp, &sub_die_temp, "Charger die temperature", NULL),
	SHELL_SUBCMD_SET_END);
```

File: drivers/npmx/npmx_shell.c [hunk 4]
```diff
@@ -520,6 +520,133 @@ static int cmd_charger_trickle_set(const struct shell *shell, size_t argc, char
...
+	npmx_charger_t *charger_instance = npmx_charger_get(npmx_instance, 0);
+	bool status;
+
+	npmx_error_t err_code = npmx_charger_die_temp_status_get(charger_instance, &status);
+
+	if (check_error_code(shell, err_code)) {
+		if (status == true) {
+			shell_print(shell, "Charger die temperature above high threshold.");
+		} else {
+			shell_print(shell, "Charger die temperature below high threshold.");
```
Comment (masz-nordic) [inline drivers/npmx/npmx_shell.c:640]: This should print `Success: %d`

File: drivers/npmx/npmx_shell.c [hunk 5]
```diff
@@ -520,6 +520,133 @@ static int cmd_charger_trickle_set(const struct shell *shell, size_t argc, char
...
+		return 0;
+	}
+
+	npmx_charger_t *charger_instance = npmx_charger_get(npmx_instance, 0);
+	uint16_t temperature;
+
+	npmx_error_t err_code = func(charger_instance, &temperature);
+
+	if (check_error_code(shell, err_code)) {
+		shell_print(shell, "Temperature threshold: %d *C.", temperature);
```
Comment (masz-nordic) [inline drivers/npmx/npmx_shell.c:591]:
```suggestion
		shell_print(shell, "Success: %d *C.", temperature);
```

**Review by masz-nordic — APPROVED** [review]

**Pull request merged**